cpp/src/arrow/util/basic_decimal.cc

```diff
@@ -949,33 +949,31 @@ DecimalStatus BasicDecimal128::Rescale(int32_t original_scale, int32_t new_scale
 
 void BasicDecimal128::GetWholeAndFraction(int scale, BasicDecimal128* whole,
                                           BasicDecimal128* fraction) const {
-  DCHECK_GE(scale, 0);
-  DCHECK_LE(scale, 38);
+  DCHECK_EQ(ValidateScale(scale), true);
 
   BasicDecimal128 multiplier(ScaleMultipliers[scale]);
   auto s = Divide(multiplier, whole, fraction);
   DCHECK_EQ(s, DecimalStatus::kSuccess);
 }
 
+bool BasicDecimal128::ValidateScale(int32_t scale) { return (std::abs(scale) <= 38); }
```

Comment:
This function can accept `-38 <= scale <= 38` while the original code accepted `0 <= scale <= 38`.   
Is this change expected? If so, it would be good to leave a comment.

Reply:
Especially as `ScaleMultipliers` is then going to be indexed with a negative value above.